crates/ruff_python_formatter/src/comments/placement.rs

```diff
@@ -1211,6 +1212,35 @@ fn handle_import_from_comment<'a>(
     }
 }
 
+/// Attach an enclosed end-of-line comment to a [`ast::StmtWith`].
+///
+/// For example, given:
+/// ```python
+/// with ( # foo
+///     CtxManager1() as example1,
+///     CtxManager2() as example2,
+///     CtxManager3() as example3,
+/// ):
+///     ...
+/// ```
+///
+/// The comment will be attached to the [`ast::StmtWith`] node as a dangling comment, to ensure
+/// that it remains on the same line as the [`ast::StmtWith`] itself.
+fn handle_with_comment<'a>(
+    comment: DecoratedComment<'a>,
+    with_: &'a ast::StmtWith,
```

Comment:
Nit: Name this `with_statement`. The `with_` gives me the impression that something is missing (also hard to see the `.` right after the underscore).